File: src/main/java/io/teamif/rpgstats/listener/CommonListener.java
package io.teamif.rpgstats.listener;

import io.teamif.rpgstats.plugin.RPGStatsPlugin;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class CommonListener implements Listener {
    private final RPGStatsPlugin plugin;

    public CommonListener(RPGStatsPlugin plugin) {
        this.plugin = plugin;
    }

    @EventHandler(ignoreCancelled = true)
    public void onPlayerJoin(PlayerJoinEvent event) {
        plugin.getLogger().info(event.getPlayer().getName() + " joined.");
    }
}

File: src/main/kotlin/io/teamif/rpgstats/listener/CommonListener.kt
package io.teamif.rpgstats.listener

import io.teamif.rpgstats.player.Player
import io.teamif.rpgstats.plugin.RPGStatsPlugin
import io.teamif.rpgstats.role.RoleType
import io.teamif.rpgstats.tribe.TribeType
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

internal class CommonListener : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        RPGStatsPlugin.IOManager.playerData.data[event.player.uniqueId.toString()] = Player(TribeType.Human, RoleType.Rogue)
    }
}